src/panel_gwalker/_gwalker.js

```diff
@@ -0,0 +1,106 @@
+import {GraphicWalker} from "graphic-walker"
+import {useEffect, useState, useRef} from "react"
+
+function transform(data) {
+  const keys = Object.keys(data);
+  const length = data[keys[0]].length;
+
+  return Array.from({ length }, (_, i) =>
+    keys.reduce((obj, key) => {
+      obj[key] = data[key][i];
+      return obj;
+    }, {})
+  );
+}
+
+function cleanToDict(value){
+    value = JSON.stringify(value)
+    value = JSON.parse(value)
+    return value
+}
+
+export function render({ model }) {
+  const [data] = model.useState('object')
+  const [fields] = model.useState('fields')
+  const [computation] = model.useState('computation')
+  const [appearance] = model.useState('appearance')
+  const [config] = model.useState('config')
+  const [currentChart, setCurrentChart] = model.useState("current_chart")
+  const [exportCurrentChart,setExportCurrentChart] = model.useState("export_current_chart")
+  const [currentChartList, setCurrentChartList] = model.useState("current_chart_list")
+  const [exportCurrentChartList,setExportCurrentChartList] = model.useState("export_current_chart_list")
+  const [payloadRequest, setPayloadRequest] = model.useState("_payload_request");
+  const [transformedData, setTransformedData] = useState([]);
+  const [computationFunc, setComputationFunc] = useState(null);
+
+  const graphicWalkerRef = useRef(null);
+
+  if (exportCurrentChart && graphicWalkerRef && graphicWalkerRef.current){
+    (async () => {
+      let value = await graphicWalkerRef.current.exportChart()
+      value=cleanToDict(value)
+      setCurrentChart(value)
+      setExportCurrentChart(false)
+    })()
+  }
+
+  if (exportCurrentChartList && graphicWalkerRef && graphicWalkerRef.current){
+    const chartList = [];
+    (async () => {
+        for await (const chart of graphicWalkerRef.current.exportChartList()) {
+            chartList.push(cleanToDict(chart))
+        }
+        setCurrentChartList(chartList)
+        setExportCurrentChartList(false)
+    })()
+  }
+
+  useEffect(() => {
+    let result = null
+    if (computation=="client"){
+      result = transform(data);
+    }
+    setTransformedData(result);
+  }, [data, computation]);
+
+  const _computationFunc = (value) => {
```

Comment:
In the _computationFunc i need to send a request with a payload to the server and return a response from the server in the same function call.

Currently I sleep and wait for the response. What is the correct way to do this @philippjfr ?

Reply:
This now works nicely on a Chart page where only one request is sent at the time. But when I navigate to the `Data` page multiple requests are sent asynchronously and then a more robust communication system is needed.

If I was to try to find a solution I would probably try to build a queueing system on the client side where the responses arrived with an id. But I hope that Panel/ Bokeh already provides something that can be used. I would probably just be wasting my time.

Could you take a look @philippjfr and try to see if you can fix the issue? Or provide some guidance on how to best solve this.

You can see in the end of the video below that something goes wrong when navigating to the data page.

https://github.com/user-attachments/assets/af7a2640-ccb9-446b-ac61-d4bef8339abb

Explore with

```python
import pandas as pd
import panel as pn

from panel_gwalker import GraphicWalker

pn.extension()

df = pd.read_csv(
    "https://datasets.holoviz.org/windturbines/v1/windturbines.csv.gz", nrows=10000
)
# df = pd.DataFrame({"a": [1, 2, 3]})
# Enable server-side computation for scalable data processing
walker = GraphicWalker(df, computation="server")

pn.Column(
    walker,
    walker.param.computation,
    walker.param._payload_request,
    walker.param._payload_response,
).servable()
```

Reply:
Definitely should be using events here, not parameter values.

Reply:
Could you try to implement it? Would not know how to do it.